### DISCUSSION TITLE
Default `cflux` / `flux` vs `sep.sum_*` methods

### DISCUSSION THREAD

**Discussion by RReverser** [sep-developers/sep#103] [Q&A]
`ctest/test_image.c` - which I assume is supposed to be a canonical usage example from C - is doing those steps in order:

 - `sep_background`
 - `sep_bkg_array`
 - `sep_bkg_subarray`
 - `sep_extract`
 - `sep_sum_circle`

However, I've noticed that on practice `sep_extract` already populates `cflux` + `flux` fields (and the documentation confirms that - https://sep.readthedocs.io/en/v0.5.x/api/sep.extract.html), so I guess my questions are, 1) which model does those default flux values correspond to and 2) correspondingly, under which circumstances I would _want_ to invoke one of the `sep_sum_*` functions instead of using those values populated by `sep_extract`?

Apologies if it's supposed to be obvious, but I haven't found an answer or mention of those fields in https://sep.readthedocs.io/en/v0.5.x/apertures.html.

**Comment by kbarbary** [comment]
Both the output of `sep.extract` and the `sep.sum_*` functions are referred to as "flux", so this is indeed confusing. The `cflux` and `flux` values from `sep.extract` are the sum of the member pixels of the object, which in general is some set of connected pixels above the detection threshold, but can have arbitrary irregular shape. The `sep.sum_*` functions are the sum of all pixels within an aperture, regardless of their value and regardless of whether they belong to an "object" or not. Does that help?

**Reply by RReverser** [reply]
Somewhat, yes, thanks... Can you provide a rough example where those values would significantly disagree? Would it be for "smudged" stars?
